### PR TITLE
Fix/get correct version from if match

### PR DESCRIPTION
**Pull request by fduch2k**
Remove starting and finishing double quotes and remove escaping from quotes in the middle

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@fduch2k__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: src/express/middleware/parseItem.js [hunk 1]
```diff
@@ -21,7 +22,7 @@ module.exports = function (table) {
                 // set version property from if-match header, if specified
                 var etag = req.get('if-match');
                 if(etag) {
-                    req.azureMobile.version = etag;
+                    req.azureMobile.version = strings.getVersionFromIfMatch(etag);
```
Comment (danderson00) [inline src/express/middleware/parseItem.js]: We will remove setting the version on the azureMobile context object. This is redundant and not required. All tests pass without it. This makes the code a bit cleaner:

    var etag = req.get('if-match');
    if(etag)
        item.version = etag;

File: src/utilities/strings.js [hunk 2]
```diff
@@ -25,6 +25,11 @@ var strings = module.exports = {
         return !stringIdValidatorRegex.test(id);
     },
 
+    // remove starting and finishing quotes and remove quote escaping from the middle of a string
+    getVersionFromIfMatch: function (value) {
```
Comment (danderson00) [inline src/utilities/strings.js]: Rename helper function to something more generic "extractEtag"

**Comment by danderson00** [comment]
Should be all that is required to resolve https://github.com/Azure/azure-mobile-apps-node/issues/45

**Comment by danderson00** [comment]
The dev branch was deleted, please make all PRs against the master branch from now on. You should be able to just recreate this PR. Sorry about that.

**Pull request closed without merging**